ethereum-common/src/calls.rs

```diff
@@ -92,3 +96,70 @@ pub fn call_keys(call: &substreams_ethereum::pb::eth::v2::Call) -> Vec<String> {
 
     keys
 }
+
+pub mod tests {
+    use super::*;
+
+    use crate::pb::sf::substreams::ethereum::v1::{Call, Calls};
+    use prost::Message;
+
+    use anyhow::Error;
+    use substreams_ethereum::pb::eth::v2::Block;
+
+    use base64::decode;
+    use std::fs;
+
+    #[test]
+    fn test_all_calls() {
+        let block = parse_block().expect("Failed to parse block");
+
+        let result = _all_calls(block).expect("Failed to execute function");
+        assert_eq!(result.calls.len(), 670);
+    }
+
+    #[test]
+    fn test_filtered_calls() {
+        // Given
+        let block = parse_block().expect("Failed to parse block");
+
+        let calls_obj = Calls {
+            calls: block
+                .calls()
+                .into_iter()
+                .map(|c| {
+                    return Call {
+                        tx_hash: "0x".to_owned(),
+                        call: Some(c.call.clone()),
+                    };
+                })
+                .collect(),
+            clock: None,
+        };
+
+        // When
+        let result = _filtered_calls(
+            "call_from:0x5acc84a3e955bdd76467d3348077d003f00ffb97".to_owned(),
+            calls_obj,
+        )
+        .expect("Failed to execute function");
+
+        // Expect
+        result.calls.iter().for_each(|c| {
+            let caller = &c.call.as_ref().unwrap().caller;
+
+            assert_eq!(
+                Hex::encode(&caller),
+                "5acc84a3e955bdd76467d3348077d003f00ffb97"
+            );
+        });
+    }
+
+    fn parse_block() -> Result<Block, Error> {
```

Comment:
No need to return `Result` in tests, everything should work, you can panic if something is wrong with `.expect(...)`.